### PR TITLE
chore(core): refactor kmx_file.h to common 🚑

### PR DESCRIPTION
**Pull request by mcdurdin**
Part of the kmcompx refactor. We want the definitions in kmx_file.h rather than continuing to use legacy_kmx_file.h.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by mcdurdin** [comment]
@ermshiperete can you help me get the Keyman packaging (Linux) build going?

**Comment by ermshiperete** [comment]
Looks like we need to add `common/include` to the source package in `linux/scripts/dist.sh`, partially reverting dbca5712e71e7c48a9e40d87cd36ccec2591e23b.

File: common/include/km_types.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#define KMX_64BIT
+#endif
+
+typedef uint32_t            KMX_DWORD;
+typedef int32_t             KMX_BOOL;
+typedef uint8_t             KMX_BYTE;
+typedef uint16_t            KMX_WORD;
+
+#if defined(__cplusplus)
+typedef char16_t   km_kbp_cp;
```
Comment (srl295) [inline common/include/km_types.h:28]: hooray

**Review by srl295 — APPROVED** [review]
LGTM FWIW

File: common/include/km_types.h (hunk 1 above)
Comment (mcdurdin) [reply common/include/km_types.h:28]: I'm not sure if I am happy or sad. `char16_t` is so unloved by the standard libraries!

Comment (srl295) [reply common/include/km_types.h:28]: use ICU4C instead!

Anyway, it's a happier type than wchar will ever be…

**Review by rc-swag — APPROVED** [review]
lgtm

File: common/include/km_types.h (hunk 1 above)
Comment (mcdurdin) [reply common/include/km_types.h:28]: Let's have a discussion on using ICU4C in the Keyman Core. I think we stand to gain considerable benefit. Only concern is the additional weight for the WASM build.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.47-alpha](https://keyman.com/downloads/releases/alpha/16.0.47)

**Pull request merged**